src/components/clusterConfiguration/NetworkConfigurationForm.tsx

```diff
@@ -0,0 +1,201 @@
+
+import NetworkConfiguration from './NetworkConfiguration';
+import ClusterSshKeyField from './ClusterSshKeyField';
+import { getHostSubnets, getNetworkInitialValues } from './utils';
+
+const validationSchema = (initialValues: NetworkConfigurationValues, hostSubnets: HostSubnets) =>
+  Yup.lazy<NetworkConfigurationValues>((values) =>
+    Yup.object<NetworkConfigurationValues>().shape({
+      baseDnsDomain: dnsNameValidationSchema(initialValues.baseDnsDomain),
+      clusterNetworkHostPrefix: hostPrefixValidationSchema(values),
+      clusterNetworkCidr: ipBlockValidationSchema,
+      serviceNetworkCidr: ipBlockValidationSchema,
+      apiVip: vipValidationSchema(hostSubnets, values, initialValues.apiVip),
+      ingressVip: vipValidationSchema(hostSubnets, values, initialValues.ingressVip),
+      sshPublicKey: sshPublicKeyValidationSchema,
+      additionalNtpSource: ntpSourceValidationSchema,
+    }),
+  );
+
+const NetworkConfigurationForm: React.FC<{
+  cluster: Cluster;
+  managedDomains: ManagedDomain[];
+}> = ({ cluster, managedDomains }) => {
+  /* TODO(mlibra): Refactor alerts section along the butons
+    const [isValidationSectionOpen, setIsValidationSectionOpen] = React.useState(false);
+  const [isStartingInstallation, setIsStartingInstallation] = React.useState(false);
+  */
+  const { onBack, onNext } = useClusterWizardTransitionFunctions(CLUSTER_STEP_NETWORKING);
+  const { alerts, addAlert, clearAlerts } = React.useContext(AlertsContext);
+  const dispatch = useDispatch();
+  const hostSubnets = React.useMemo(() => getHostSubnets(cluster), [cluster]);
+  const initialValues = React.useMemo(() => getNetworkInitialValues(cluster, managedDomains), [
+    cluster,
+    managedDomains,
+  ]);
+
+  const memoizedValidationSchema = React.useMemo(
+    () => validationSchema(initialValues, hostSubnets),
+    [hostSubnets, initialValues],
+  );
+
+  const handleSubmit = async (
+    values: NetworkConfigurationValues,
+    formikActions: FormikHelpers<NetworkConfigurationValues>,
+  ) => {
+    clearAlerts();
+
+    // update the cluster configuration
+    try {
+      const params = _.omit(values, ['hostSubnet', 'useRedHatDnsService', 'shareDiscoverySshKey']);
+
+      // Discard additionalNtpSource if it is empty
+      if (!values.additionalNtpSource) {
+        delete params.additionalNtpSource;
+      }
+
+      if (values.shareDiscoverySshKey) {
+        params.sshPublicKey = cluster.imageInfo.sshPublicKey;
+      }
+
+      if (values.vipDhcpAllocation) {
+        delete params.apiVip;
+        delete params.ingressVip;
+        const cidr = hostSubnets.find((hn) => hn.humanized === values.hostSubnet)?.subnet;
+        params.machineNetworkCidr = cidr;
+      }
+
+      const { data } = await patchCluster(cluster.id, params);
+      formikActions.resetForm({
+        values: getNetworkInitialValues(data, managedDomains),
+      });
+      dispatch(updateCluster(data));
+
+      if (canNextNetwork({ cluster })) {
```

Comment:
This is the example that trying to make `canNextNetwork` reusable is confusing. `canNextNetwork` is checking if form is submitting and valid but this time we're passing `cluster` prop to it. So we're trying to have a single function doing 2 very different things. I'd much rather prefer this here:

```suggestion
      if (getClusterHardwareStatus(cluster) === 'ready') {
        setCurrentStepId('networking')
        ...
      
```

If there are more things to check, we wrap it in
```
canTransitionToNextStep = (cluster) => boolean
```